WordPress/Classes/Models/Notifications/NotificationRange.swift

```diff
@@ -79,6 +82,30 @@ class NotificationRange {
             break
         }
     }
+
+
+    /// Returns the NotificationRange Kind, for a given raw Notification Range.
+    ///
+    /// - Details:
+    ///     I truly hope the reviewer, and the Opensource Community can forgive this ongoing hack.
+    ///     Notifications can now carry URL's without specifying an explicit 'Type'. For that reason,
+    ///     surprise!... we're now also inferring the Notification Type.
+    ///
+    private static func kind(for dictionary: [String: AnyObject]) -> Kind? {
+        if let type = dictionary[RangeKeys.RawType] as? String, let kind = Kind(rawValue: type) {
+            return kind
+        }
+
+        if dictionary[RangeKeys.SiteId] != nil {
+            return .Site
+        }
+
+        if dictionary[RangeKeys.URL] != nil {
```

Comment:
The `nil` check is fine by me but would it be swiftier to have something like
`if _ = dictionary[RangeKeys.URL]`